Dockerfile

```diff
@@ -1,15 +1,37 @@
-FROM golang:alpine
+ARG GO_VERSION=1.17
+ARG ALPINE_VERSION=3.15
+
+# Build stage
+FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as build
 
 WORKDIR /app
 
-COPY go.mod .
-COPY go.sum .
+COPY go.mod go.sum .
+
 RUN go mod download
 
-COPY . .
+COPY cmd cmd
+
+COPY server server
```

Comment:
Why split these?

Reply:
`COPY cmd server .` wouldn't work as intended (but would behave like `COPY cmd/* server/* .` somehow.

Reply:
Sorry, I mean why split `COPY . .` into `cmd`, `server`, etc.?

Reply:
I edited it my previous message so maybe you've missed it:

> Also, the idea to replace COPY . . was that any changed file (even an irrelevant change to the build) would trigger a re-build, which was not efficient.

The idea was that the build should only be triggered when necessary. You want to have cached layers when testing small changes that shouldn't trigger a full re-build.